cirq/sim/state.py

```diff
@@ -121,3 +123,157 @@ def validate_normalized_state(state: np.ndarray, num_qubits: int,
     norm = np.sum(np.abs(state) ** 2)
     if not np.isclose(norm, 1):
         raise ValueError('State is not normalized instead had norm %s' % norm)
+
+
+def sample_state_vector(
+    state: np.ndarray,
+    indices: List[int],
+    repetitions: int=1) -> List[List[bool]]:
+    """Samples repeatedly from measurements in the computational basis.
+
+    Note that this does not modify the passed in state.
+
+    Args:
+        state: The state to be measured. This state is assumed to be normalized.
+            The state must be of size 2 ** integer.  The state can be of shape
+            (2 ** integer) or (2, 2, ..., 2).
+        indices: Which qubits are measured. The state is assumed to be supplied
+            in big endian order. That is the xth index of v, when expressed as
+            a bitstring, has the largest values that the 0th index.
+        repetitions: The number of times to sample the state.
+
+    Returns:
+        Measurement results with True corresponding to the |1> state.
+        The outer list is for repetitions, and the inner corresponds to
+        measurements ordered by the input indices.
+
+    Raises:
+        ValueError if repetitions is less than one or size of state is not a
+            power of 2.
+        IndexError if the indices are out of range for the number of qubits
+            corresponding to the state.
+    """
+    if repetitions < 1:
+        raise ValueError(
+            'Number of repetitions cannot be negative. Was {}'.format(
```

Comment:
Message doesn't match the check (0 isn't negative).

I think repetitions==0 should be allowed and explicitly tested for.

Reply:
I always disagree about this.  It is a common bug to end up with a do-nothing case that you don't realize because you are passing in something that is not correct and you spend lots of time before you notice this side effect.  The counter is that you should handle these edge cases so that code that hits this as an edge case is fine.  I find the later less compelling, I'd rather have my code yell at me and save me from the pain of the later than counter me not being able handle my edge cases correctly or explicitly.  

But I changed it because I'm generous that way.